### PR TITLE
Fix Edit Form for Individual Events

### PR DESCRIPTION
**Pull request by xuebingli**
This PR proposes two fixes to the edit form for individual events

- Fix form population for individual event edit form. #23 
- Fix optimistic UI for the edit form. Previously, the newly (created *or edited*) event is always appended to the list of existing events. This leads to duplicate events in the list in the case of edits.

### CODE REVIEW

File: app/javascript/pages/IndividualEvents/index.js [hunk 1]
```diff
@@ -378,6 +379,22 @@ const individualEventInputToOptimisticResponse = (data, input) => {
   }
 }
 
+const buildOptimisticResponse = (newEvent, currentUser) => {
+  const { id: currentUserId, individualEvents: existingEvents } = currentUser
+  const arrIdx = R.find(R.propEq('id', newEvent.id), existingEvents)
+  const individualEvents = R.isNil(arrIdx)
+    ? R.append(newEvent, existingEvents)
+    : R.update(arrIdx, newEvent, existingEvents)
+  return {
```
Comment (craig-day) [inline app/javascript/pages/IndividualEvents/index.js]: would prefer a newline above the return, easier to read IMO

File: app/javascript/pages/IndividualEvents/index.js [hunk 2]
```diff
@@ -393,14 +410,7 @@ const withData = compose(
...
           variables: { input: individualEventInput },
-          optimisticResponse: {
-            __typename: 'Mutation',
-            createEditIndividualEvent: {
-              __typename: 'User',
-              id: currentUser.id,
-              individualEvents: R.append(newEvent, currentUser.individualEvents),
-            },
-          },
+          optimisticResponse: buildOptimisticResponse(newEvent, currentUser),
```
Comment (craig-day) [inline app/javascript/pages/IndividualEvents/index.js]: was this extraction just for readability here? I don't see it being reused

File: app/javascript/pages/IndividualEvents/index.js (hunk 1 above)
Comment (xuebingli) [reply app/javascript/pages/IndividualEvents/index.js]: Done.

File: app/javascript/pages/IndividualEvents/index.js (hunk 2 above)
Comment (xuebingli) [reply app/javascript/pages/IndividualEvents/index.js]: Yes, for improved readability and consistency https://github.com/zendesk/volunteer_portal/search?q=buildOptimisticResponse&unscoped_q=buildOptimisticResponse

**Comment by xuebingli** [comment]
@craig-day ping

**Review by craig-day — APPROVED** [review]

**Pull request merged**